src/Api/HL/Search.php

```diff
@@ -108,15 +109,16 @@ private function getSelectCriteriaForProperty(string $prop_name, bool $distinct_
                 if (str_contains($sql_field, '.')) {
                     $join_name = explode('.', $sql_field, 2)[0];
                     if (array_key_exists($join_name, $this->joins) && $this->joins[$join_name]['parent_type'] === 'array') {
-                        $expression = 'IFNULL(' . $expression . ', 0x0)';
+                        $expression = QueryFunction::ifnull($sql_field, new QueryExpression($DB::quoteValue('0x0')));
                         if ($distinct_groups) {
-                            $expression = "GROUP_CONCAT(DISTINCT $expression SEPARATOR 0x1D)";
+                            $expression = QueryFunction::groupConcat($expression, '0x1D', true);
```

Comment:
To prevent automatic quoting of separator.
```suggestion
                            $expression = QueryFunction::groupConcat($expression, new QueryExpression('0x1D'), true);
```